lib/country-bounding-boxes.ts

```diff
@@ -0,0 +1,217 @@
+  Singapore: [103.62, 1.28, 104.0, 1.36],
+  Sudan: [21.94, 8.62, 38.41, 22.0],
+  "South Sudan": [23.89, 3.51, 35.3, 12.25],
+  Senegal: [-17.63, 12.33, -11.47, 16.6],
+  "Solomon islands": [156.49, -10.83, 162.4, -6.6],
+  "Sierra Leone": [-13.25, 6.79, -10.23, 10.05],
+  "El Salvador": [-90.1, 13.15, -87.72, 14.42],
+  Somalia: [40.98, -1.68, 51.13, 12.02],
+  Serbia: [18.83, 42.25, 22.99, 46.17],
+  Suriname: [-58.04, 1.82, -53.96, 6.03],
+  Slovakia: [16.88, 47.76, 22.56, 49.57],
+  Slovenia: [13.7, 45.45, 16.56, 46.85],
+  Sweden: [11.03, 55.36, 23.9, 69.11],
+  Swaziland: [30.68, -27.29, 32.07, -25.66],
+  Syria: [35.7, 32.31, 42.35, 37.23],
+  Chad: [13.54, 7.42, 23.89, 23.41],
+  Togo: [-0.05, 5.93, 1.87, 11.02],
+  Thailand: [97.38, 5.69, 105.59, 20.42],
+  Tajikistan: [67.44, 36.74, 74.98, 40.96],
+  Turkmenistan: [52.5, 35.27, 66.55, 42.75],
+  "East Timor": [124.97, -9.39, 127.34, -8.27],
+  "Trinidad and Tobago": [-61.95, 10.0, -60.9, 10.89],
+  Tunisia: [7.52, 30.31, 11.49, 37.35],
+  Turkey: [26.04, 35.82, 44.79, 42.14],
+  Taiwan: [120.11, 21.97, 121.95, 25.3],
+  Tanzania: [29.34, -11.72, 40.32, -0.95],
+  Uganda: [29.58, -1.44, 35.04, 4.25],
+  Ukraine: [22.09, 44.36, 40.08, 52.34],
+  Uruguay: [-58.43, -34.95, -53.21, -30.11],
+  USA: [-125.0, 25.0, -66.96, 49.5],
+  Uzbekistan: [55.93, 37.14, 73.06, 45.59],
+  Venezuela: [-73.3, 0.72, -59.76, 12.16],
+  Vietnam: [102.17, 8.6, 109.34, 23.35],
+  Vanuatu: [166.63, -16.6, 167.84, -14.63],
+  "West Bank": [34.93, 31.35, 35.55, 32.53],
+  Yemen: [42.6, 12.59, 53.11, 19.0],
+  "South Africa": [16.34, -34.82, 32.83, -22.09],
+  Zambia: [21.89, -17.96, 33.49, -8.24],
+  Zimbabwe: [25.26, -22.27, 32.85, -15.51],
+  Micronesia: [134.2, 6.89, 180.0, 21.68],
+  "French Polynesia": [-164.7, -27.6, -133.9, -7.1],
+  "Saint Kitts and Nevis": [-62.86, 17.08, -62.53, 17.44],
+  Seychelles: [52.98, -7.08, 57.53, -3.37],
+  "Northern Tanzania": [29.34, -11.72, 40.32, -0.95],
+  "French Guiana": [-54.5, 2.16, -51.6, 5.77],
+  "St Lucia": [-61.07, 13.72, -60.85, 14.1],
+};
+
+export const getBoundingBoxFromCountryName = (
+  countryName: string
+): CountryBoundingBox | undefined => {
+  return countryBoundingBoxData[countryName];
+};
+
+export const combineTwoCountryBoundingBoxes = (
+  boundingBoxA: CountryBoundingBox,
+  boundingBoxB: CountryBoundingBox
+): CountryBoundingBox => {
+  const [boxAMinLng, boxAMinLat, boxAMaxLng, boxAMaxLat] = boundingBoxA;
+  const [boxBMinLng, boxBMinLat, boxBMaxLng, boxBMaxLat] = boundingBoxB;
+
+  return [
+    Math.min(boxAMinLng, boxBMinLng),
+    Math.min(boxAMinLat, boxBMinLat),
+    Math.max(boxAMaxLng, boxBMaxLng),
+    Math.max(boxAMaxLat, boxBMaxLat),
+  ];
+};
+
+export const combineCountryBoundingBoxes = (
+  boundingBoxes: CountryBoundingBox[]
+): CountryBoundingBox => {
+  return boundingBoxes.reduce((accumulator, currentValue) =>
+    combineTwoCountryBoundingBoxes(accumulator, currentValue)
+  );
+};
```

Comment:
All of this code here is so that when you pick more than one country it tries it's best to fit both of them in one box. If the box is too big it just resets the position of the map.